notebooks/MAST/astroqueryMAST/beginner_program_id_search/beginner_program_id_search.ipynb

```diff
@@ -0,0 +1,402 @@
+   "metadata": {},
+   "source": [
+    "This produces over 3000 data products associated with this observation! This is not uncommon for a JWST level-3 observation. In the next section, we'll take a look at how we can filter down the number of results before we download them."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Filtering and Downloading Data\n",
+    "\n",
+    "### Filtering"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can apply filter keyword arguments to download only data products that meet your given criteria. Available filters are “mrp_only” ([minimum recommended products](https://outerspace.stsci.edu/display/MASTDOCS/Minimum+Recommended+Products)), “extension” (file extension), calib_level (calibration level), and all products fields listed [here](https://mast.stsci.edu/api/v0/_productsfields.html).\n",
+    "\n",
+    "In this example, let's try filtering for only the level 2, calibrated exposures. It is important that we also filter by \"SCIENCE\" type products; otherwise, our results will include guide star acquisition images."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filtered_prod = Observations.filter_products(data_products, calib_level=[2], productType=\"SCIENCE\")\n",
+    "\n",
+    "# Again, we choose columns of interest for convenience\n",
+    "disp_col = ['obsID','dataproduct_type','productFilename','size','calib_level']\n",
+    "filtered_prod[disp_col].show_in_notebook(display_length=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Well, that was effective! We now have 80 files, instead of over 3000.\n",
+    "\n",
+    "As a final check before we proceed to the download, let's find the total file size of our results:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "total = 0\n",
+    "for fsize in filtered_prod['size']:\n",
+    "    total = total+fsize\n",
+    "print('{:.2f}'.format(total/10**9), \"GB\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For downloads larger than a GB, it is highly recommended that you follow the steps in [Downloading via Curl Script](#Downloading-via-Curl-Scipt) rather than attempting to download the data directly."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Downloading Data Directly\n",
+    "\n",
+    "We'll use the filtered product list to select our downloads. This method will immediately send a request to the MAST archives, and download the data to this notebook's folder.\n",
+    "\n",
+    "**Note**: By default, this will only download the first five files. This reduces download time for the purposes of the tutorial, but still demonstrates a successful download."
```

Comment:
Include explicit instructions on how to modify the script to download all the products (e.g., "remove the [:5] piece from the line below", or some such instruction.